File: runme/mcgen/local/datan/datsrc/rnradi.c
#include <math.h>
#include "datsrc.h"

void LIBFUNC rnradi_(double const P_T *a, double const P_T *tau1,
                    double const P_T *tau2, double P_T *t)
{
    /* Table of constant values */

    integer c__3 = 3;

    /* Local variables */
    double r[3];

    rnecuy_(r, &c__3);
    if (r[0] < *a) {
/* Mean life is TAU1 */
        *t = -(*tau1) * log(r[1]);
    } else {
/* Mean life is TAU2 */
        *t = -(*tau2) * log(r[2]);
    }
    return;
} /* rnradi_ */

File: runme/mcgen/local/datan/datsrc/rnradi.f
      SUBROUTINE RNRADI(A,TAU1,TAU2,T)
      IMPLICIT DOUBLE PRECISION (A-H,O-Z)
      DIMENSION R(3)
      CALL RNECUY (R,3)
      IF(R(1).LT.A) THEN
C Mean life is TAU1
        T=-TAU1*LOG(R(2))
      ELSE
C Mean life is TAU2
        T=-TAU2*LOG(R(3))
      END IF
      END
